torch/csrc/api/include/torch/enum.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <string>
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+#define TORCH_ENUM_DECLARE(name) \
+namespace torch { \
+namespace enumtype { \
+  /*
+    NOTE: We need to provide the default constructor for each struct,
+    otherwise Clang 3.8 would complain:
+    ```
+    error: default initialization of an object of const type 'const enumtype::Enum1'
+    without a user-provided default constructor
+    ```
+  */ \
+  struct k##name { k##name() {} }; \
+} \
+TORCH_API extern const enumtype::k##name k##name; \
+}
+
+#define TORCH_ENUM_DEFINE(name) \
+namespace torch { \
+const enumtype::k##name k##name; \
+}
+
+#define TORCH_ENUM_PRETTY_PRINT(name) \
+std::string operator()(enumtype::k##name& v) const { \
```

Comment:
You can return a `const char*` here right? Seems better.

Reply:
Fixed